receiver.py

```diff
@@ -63,16 +70,58 @@ def async_tx(self, command):
         """
         self.outbound.append(command)
 
-    def __enter__(self):
-        #Change this to search for USB, Unhardcode the ports
-        if _platform == "linux" or _platform == "linux2":
-            self.ser = serial.Serial('/dev/ttyUSB0', 115200)
-        elif _platform == "win32":
-            self.ser = serial.Serial('COM5', 115200)
-        self.xbee = ZigBee(self.ser)
-        print 'xbee created/initialized'
+
+    def reconnect_xbee(self):
+
+        #detect platform and format port names
+        if _platform.startswith('win'):
+            ports = ['COM%s' % (i + 1) for i in range(256)]
+        elif _platform.startswith('linux'):
+            # this excludes your current terminal "/dev/tty"
+            ports = glob.glob('/dev/tty[A-Za-z]*')
+        else:
+            raise EnvironmentError('Unsupported platform: ' + _platform)
+
+
+
+        #search for available ports
+        port_to_connect = ''
+        while port_to_connect == '': 
+            ports_avail = []
+            
+            #loop through all possible ports and try to connect
+            for port in ports:
+                try:
+                    s = serial.Serial(port)
+                    s.close()
+                    ports_avail.append(port)
+                except (OSError, serial.SerialException):
+                    pass
+            
+            if len(ports_avail) ==1:
+                port_to_connect = ports_avail[0]
+                
+            elif len(ports_avail)==0:
+                #No Serial port found, continue looping.
+                print "No serial port detected. Trying again..."
+                time.sleep(1)
+
+            elif len(ports_avail)>1:
+                #Multiple serial ports detected. Get user input to decide which one to connect to
+                com_input = raw_input("Multiple serial ports available. Which serial port do you want? \n"+str(self.ports_avail)).upper()+":";
```

Comment:
I suppose this is a way to handle this, but if you are going to require interaction anyway, you might as well specify the port to use on command line (which is probably a good idea anyway).